amulet/deployer.py

```diff
@@ -330,6 +337,50 @@ def deploy_w_timeout_and_dir(self, timeout, deploy_dir):
             yield
         self.deployed = True
 
+    def action_defined(self, service):
+        """
+        :param service: Name of service to get list of defined actions for.
+
+        Returns the list of actions defined for the service.
+        """
+        if service not in self.services:
+            raise ValueError('Service needs to be added before listing actions.')
+        raw = juju(['action', 'defined', service, '--format', 'json'])
+        return json.loads(raw)
+
+    def action_do(self, unit, action, action_args={}):
+        """
+        :param unit: Unit to run action on.
+        :param action: Action to run.
+        :param action_args: Action parameters.
+
+        Runs specified action on specified unit and returns the uuid to fetch results by.
+        """
+        if '/' not in unit:
+            raise ValueError('%s is not a unit' % unit)
+        cmd = ['action', 'do', unit, action, '--format', 'json']
+        for key, value in action_args.items():
+            cmd += ["%s=%s" % (str(key), str(value))]
+        result = juju(cmd)
+        action_result = json.loads(result)
+        results_id = action_result["Action queued with id"]
+        return results_id
+
+    def action_fetch(self, action_id, wait=True, timeout=600):
```

Comment:
Having both a wait and a timeout seems odd, I'd just have 

```
if timeout is not None:
    cmd += ["--wait", str(timeout)]
```

That way a mix of options won't produce uncertain output